### DISCUSSION TITLE
My project wont deploy if i have vercel.json file.

### DISCUSSION THREAD

**Discussion by qhuge** [vercel/community#1500] [Help] [answered]
### Summary

I have no idea why, but if my project has the vercel.json file, it just doesn't deploy. In GitHub, it says "All checks have failed". I'm trying to deploy a normal static html website and i would like to set some headers on some routes. 

Here's my vercel.json.
```
{
    "headers": [
      {
        "source": "/someroute",
        "headers": [
          { "key": "header", "value": "value"},
        ]
      }
    ]
  }
```
When i delete the file, it starts deploying successfully again.
Tried it with empty vercel.json, but that also didn't work.

Am i doing something incorrectly, or am I just dumb?

**Comment by swarnava** [comment]
Do you have a sample deployment which we can take a look?

**Reply by qhuge** [reply]
I made this basic repo: https://github.com/hukeboi/vercel-thing
Have I written something incorrectly in the vercel.json file?
Added just a basic html file and it deployed. Once i added vercel.json file, it just doesn't deploy.

**Comment by qhuge** [comment, marked as answer]
Oh nvm I'm just dumb. When i look at the file in GitHub, it shows a \ufeff character at the start of the vercel.json file. I removed it and it works. I have no idea how that got there, since vscode doesn't show any characters there. Probably something to do with the encoding settings or something. idk

**Comment by hassanprodeveloper** [comment]
I am also getting this error,
I don't have any \ufeff character in my vercel.json,

**Reply by swarnava** [reply]
Please create a new thread with reproducible repository so we can take a look!

**Reply by selalipop** [reply]
Running into a similar situation, apart from a reproducible repository for my specific case it looks like invalid vercel.json files result in deployment failures with no visible logs, is that right?

**Comment by Rodrigo-JM** [comment]
Also getting this error. 
Running Next v10, monorepo (not turborepo), and also pushing code to the development branch, not production.

**Reply by Rodrigo-JM** [reply]
Oh nvm I'm just dumb (2). I was trying to implement the redirect config but got confused in the docs and used "rewrites" instead. Since there are no logs for this error, maybe rewrites don't work if you don't have the source route
